src/main/scala/org/constellation/p2p/Cluster.scala

```diff
@@ -437,7 +437,7 @@ class Cluster[F[_]: Concurrent: Timer: ContextShift](
 
         _ <- if (dao.seedsPath.nonEmpty && dao.peersInfoPath.isEmpty) {
           C.evalOn(ConstellationExecutionContext.bounded)(
-            Sync[F].delay(Download.download()(dao, ConstellationExecutionContext.bounded))
+            LiftIO[F].liftIO(dao.downloadService.download())
```

Comment:
We have a circular reference, here and within DownloadService which has a ref to Cluster.